test/unit/org/apache/cassandra/index/sai/analyzer/AnalyzerEqOperatorSupportTest.java

```diff
@@ -0,0 +1,368 @@
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v = 'lazy'");
+    }
+
+    private void assertTokenizedIndexSupportsMatches()
+    {
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'fox'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick fox' OR v : 'Lazy fox'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick fox' OR v : 'lazy fox'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' OR v : 'Lazy'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' OR v : 'lazy'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' AND v : 'fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' AND v : 'fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' AND v : 'Lazy'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' AND v : 'lazy'");
+    }
+
+    private void assertTokenizedIndexSupportsMixedEqualityAndMatches()
+    {
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick fox' OR v : 'Lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick fox' OR v : 'lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' OR v : 'Lazy'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' OR v : 'lazy'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v : 'fox'", row(1));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v : 'fox'", row(1));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v : 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v : 'lazy'");
+
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick fox' OR v = 'Lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick fox' OR v = 'lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' OR v = 'Lazy'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' OR v = 'lazy'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' AND v = 'fox'", row(1));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' AND v = 'fox'", row(1));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' AND v = 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' AND v = 'lazy'");
```

Comment:
`assertRowsWithWarning("SELECT k FROM %s WHERE (v = 'quick' AND v : 'fox') OR v = 'dog'", row(1));`